internal/datastore/memdb/memdb.go

```diff
@@ -348,93 +34,199 @@ func NewMemdbDatastore(
 	watchBufferLength uint16,
 	revisionQuantization,
 	gcWindow time.Duration,
-	simulatedLatency time.Duration,
 ) (datastore.Datastore, error) {
 	if revisionQuantization > gcWindow {
-		return nil, fmt.Errorf(
-			errUnableToInstantiateTuplestore,
-			errors.New("gc window must be larger than quantization interval"),
-		)
+		return nil, errors.New("gc window must be larger than quantization interval")
 	}
 
-	db, err := memdb.NewMemDB(schema)
-	if err != nil {
-		return nil, fmt.Errorf(errUnableToInstantiateTuplestore, err)
+	if revisionQuantization <= 1 {
+		revisionQuantization = 1
```

Comment:
Having something like this in the datastore package removes decimal.Decimal from being directly in datastore implementations.

```go
func QuantizationFromDuration(d, gcWindow time.Duration) Revision {
	if d > gcWindow {
		return nil, fmt.Errorf(
			errUnableToInstantiate,
			fmt.Errorf("cluster gc window is less than requested gc window %d < %d",
				clusterTTLNanos,
				gcWindowNanos,
			),
		)
	}

	if d <= 1 {
		d = 1
	}

	return decimal.NewFromInt(d.Nanoseconds())
}
```

Reply:
Let's hold off on simplification, the PR in question doesn't introduce decimal.Decimal to the implementations